src/main/java/mate/academy/springboot/datajpa/repository/specification/product/ProductCategoryInSpecification.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.springboot.datajpa.repository.specification.product;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.JoinType;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductCategoryInSpecification implements SpecificationProvider<Product> {
+    private static final String FILTER_KEY = "categoryIn";
+    private static final String FIELD_NAME = "name";
+
+    @Override
+    public Specification<Product> getSpecification(String[] categories) {
+        return (root, query, cb) -> {
+            Join<Product, Category> join = root.join("category", JoinType.INNER);
+            CriteriaBuilder.In<String> predicate = cb.in(join.get(FIELD_NAME));
+            for (String value : categories) {
+                predicate.value(value);
+            }
+            return cb.and(predicate, predicate);
+        };
```

Comment:
I do not think we need 'join' here. Product has Category field. 
`FIELD_NAME = "category"; `will help